browser/extensions/api/brave_rewards_api.cc

```diff
@@ -455,5 +455,41 @@ void BraveRewardsRefreshPublisherFunction::OnRefreshPublisher(
                        std::make_unique<base::Value>(publisher_key)));
 }
 
+BraveRewardsGetAllNotificationsFunction::
+~BraveRewardsGetAllNotificationsFunction() {
+}
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetAllNotificationsFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+    RewardsServiceFactory::GetForProfile(profile);
+
+  auto list = std::make_unique<base::ListValue>();
+
+  if (!rewards_service) {
+    return RespondNow(OneArgument(std::move(list)));
+  }
+
+  auto notifications = rewards_service->GetAllNotifications();
+
+  for (auto const& item : notifications) {
```

Comment:
Small nit would it not make sense to use ‘notification‘ instead of ‘item’ and the dictionary item uses a different name?

Reply:
done